### PR TITLE
ARK Flow Basic Functionality

### PR DESCRIPTION
**Pull request by AlexKlimaj**
image

**Describe problem solved by this pull request**
This PR adds basic support for the [ARK Flow](https://arkelectron.com/product/ark-flow/). The ARK Flow is a UAVCAN Node running on an STM32F412 with a PAW3902 optical flow sensor, Broadcom AFBR-S50LV85D time-of-flight distance sensor, and Bosch BMI088 IMU. The ARK Flow is available for pre-order now. [Schematic and BOM are open source](https://github.com/ARK-Electronics/ARK_Flow). 

**Describe your solution**
A new board category for ARK boards and a `ark_can-flow` build target. The PAW3902 and BMI088 are functional. The driver for the AFBR-S50LV85D will be added in the future. 

**Test data / coverage**
The firmware was tested and verified using the nut shell. CAN communication was verified with a Pixhawk 4 Mini running current master. 

PX4 booting on ARK Flow.
```
nsh: sercon: command not found
HW arch: ARK_CAN_FLOW
FW git-hash: 06ec1d225f494e0c9374934fae0081369b8adfd8
FW version: 1.11.0 c0 (17498304)
FW git-branch: ark-flow
OS: NuttX
OS version: Release 10.0.0 (167772415)
OS git-hash: e7701b9e792e310112d0b45419f5f30fb384bca2
Build datetime: Jan 27 2021 19:20:34
Build uri: localhost
Toolchain: GNU GCC, 9.3.1 20200408 (release)
PX4GUID: 0001000000002030365459535017004c005a
MCU: STM32F???, rev. ?
nsh: mtd: command not found
Board defaults: /etc/init.d/rc.board_defaults
nsh: rgbled: command not found
nsh: rgbled_ncp5623c: command not found
nsh: rgbled_pwm: command not found
Board sensors: /etc/init.d/rc.board_sensors
INFO  [paw3902] using yaw rotation 0.000 degrees (0.000 radians)
paw3902 #0 on SPI bus 1 (devid=0x6d)
bmi088 #0 on SPI bus 1 (devid=0x6a)
bmi088 #1 on SPI bus 1 (devid=0x66)
ERROR [param] Parameter SENS_EN_PX4FLOW not found
INFO  [uavcannode] Node ID 120, bitrate 1000000

NuttShell (NSH) NuttX-10.0.0
nsh>
```

```
 PID COMMAND                   CPU(ms) CPU(%)  USED/STACK PRIO(BASE) STATE FD
   0 Idle Task                   42424 75.073   176/  512   0 (  0)  READY  3
   1 hpwork                          0  0.000   332/ 1268 249 (249)  w:sig  3
   2 lpwork                          0  0.000   564/ 1620  50 ( 50)  w:sig  3
   3 init                            0  0.000  1524/ 2612 100 (100)  w:sem  3
   4 wq:manager                      0  0.000   396/ 1260 255 (255)  w:sem  3
  17 top                            45  1.476  1468/ 4084 237 (237)  RUN    3
  11 wq:SPI1                       494 15.115  1288/ 2340 253 (253)  w:sem  3
  16 wq:uavcan                     172  5.287  1340/ 2180 236 (236)  READY  3

Processes: 8 total, 3 running, 5 sleeping, max FDs: 12
CPU usage: 21.88% tasks, 3.05% sched, 75.07% idle
Uptime: 54.340s total, 42.424s idle
```

```
nsh> listener optical_flow

TOPIC: optical_flow
 optical_flow_s
        timestamp: 109953275  (0.009109 seconds ago)
        pixel_flow_x_integral: -0.0060
        pixel_flow_y_integral: -0.0040
        gyro_x_rate_integral: nan
        gyro_y_rate_integral: nan
        gyro_z_rate_integral: nan
        ground_distance_m: 0.0000
        integration_timespan: 15880
        time_since_last_sonar_update: 0
        max_flow_rate: 7.4000
        min_ground_distance: 0.0800
        max_ground_distance: 30.0000
        frame_count_since_last_readout: 2
        gyro_temperature: 0
        sensor_id: 0
        quality: 115
nsh>
```

uavcan status on a Pixhawk 4 Mini
```

nsh> uavcan status
Pool allocator status:
        Capacity hard/soft: 500/250 blocks
        Reserved:  8 blocks
        Allocated: 4 blocks

UAVCAN node status:
        Internal failures: 0
        Transfer errors:   2
        RX transfers:      71324
        TX transfers:      3370

CAN1 status:
        HW errors: 13
        IO errors: 13
        RX frames: 281956
        TX frames: 2247

Sensor 'flow':
devname: /dev/flow
channel 0: node id 120 --> class instance 0
channel 1: empty
channel 2: empty
channel 3: empty

Online nodes (Node ID, Health, Mode):
         120 OK         OPERAT

uavcan: cycle time: 619672 events, 26895279us elapsed, 43.40us avg, min 10us max 1789us 77.684us rms
uavcan: cycle interval: 619672 events, 1813.84us avg, min 18us max 138945us 1305.569us rms
```

Optical Flow data on main Pixhawk 4 Mini
```
nsh> listener optical_flow

TOPIC: optical_flow
 optical_flow_s
        timestamp: 1224790878  (0.004232 seconds ago)
        pixel_flow_x_integral: -0.0140
        pixel_flow_y_integral: 0.0100
        gyro_x_rate_integral: nan
        gyro_y_rate_integral: nan
        gyro_z_rate_integral: 0.0000
        ground_distance_m: 0.0000
        integration_timespan: 15880
        time_since_last_sonar_update: 0
        max_flow_rate: 5.0000
        min_ground_distance: 0.1000
        max_ground_distance: 30.0000
        frame_count_since_last_readout: 0
        gyro_temperature: 0
        sensor_id: 120
        quality: 99
```

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Awesome!

File: boards/ark/can-flow/src/spi.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline boards/ark/can-flow/src/spi.cpp:3]:
```suggestion
 *   Copyright (C) 2021 PX4 Development Team. All rights reserved.
```

File: boards/ark/can-flow/src/led.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2013 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline boards/ark/can-flow/src/led.c:3]:
```suggestion
 *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```

File: boards/ark/can-flow/src/init.c [hunk 3]
```diff
@@ -0,0 +1,149 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline boards/ark/can-flow/src/init.c:3]:
```suggestion
 *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```

File: boards/ark/can-flow/src/i2c.cpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline boards/ark/can-flow/src/i2c.cpp:3]:
```suggestion
 *   Copyright (C) 2021 PX4 Development Team. All rights reserved.
```

File: boards/ark/can-flow/src/can.c [hunk 5]
```diff
@@ -0,0 +1,130 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 2012 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline boards/ark/can-flow/src/can.c:3]:
```suggestion
 *   Copyright (C) 2021 PX4 Development Team. All rights reserved.
```

File: boards/ark/can-flow/src/boot_config.h [hunk 6]
```diff
@@ -0,0 +1,136 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2012-2015 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline boards/ark/can-flow/src/boot_config.h:3]:
```suggestion
 *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```

File: boards/ark/can-flow/src/boot.c [hunk 7]
```diff
@@ -0,0 +1,199 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2015 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline boards/ark/can-flow/src/boot.c:3]:
```suggestion
 *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```

File: boards/ark/can-flow/src/board_config.h [hunk 8]
```diff
@@ -0,0 +1,94 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline boards/ark/can-flow/src/board_config.h:3]:
```suggestion
 *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```

File: boards/ark/can-flow/src/Makefile [hunk 9]
```diff
@@ -0,0 +1,100 @@
+############################################################################
```
Comment (dagar) [inline boards/ark/can-flow/src/Makefile:1]: This file can be deleted.

**Pull request merged**